src/App.tsx

```diff
@@ -1,24 +1,131 @@
-/* eslint-disable max-len */
 /* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import { UserWarning } from './UserWarning';
+import { Header } from './components/Header';
+import { Footer } from './components/Footer';
+import { Todo } from './components/Todo';
+import { Error } from './components/Error';
+import { Todo as TodoType } from './types/Todo';
+import { addTodo, deleteTodo, getTodos } from './api/todos';
+import { Filter } from './types/Filter';
 
-const USER_ID = 0;
+export const USER_ID = 10583;
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<TodoType[]>([]);
+  const [tempTodos, setTempTodos] = useState<TodoType[]>([]);
+  const [error, setError] = useState<string>('');
+  const [isInputLocked, setIsInputLocked] = useState(false);
+  const [filter, setFilter] = useState<Filter>('all');
+
+  useEffect(() => {
+    getTodos(USER_ID)
+      .then((data) => setTodos(data))
+      .catch(() => setError('Unable to fetch data'));
+  }, []);
+
+  useEffect(() => {
+    setTimeout(() => {
+      setError('');
+    }, 3000);
+  }, [error]);
+
+  useEffect(() => {
+    tempTodos.forEach((todo) => addTodo({
+      title: todo.title,
+      completed: todo.completed,
+      userId: USER_ID,
+    })
+      .then((res) => {
+        if (res) {
+          const newTodo = { ...todo };
+
+          setTempTodos((prev) => prev.filter((task) => task.id !== todo.id));
+          newTodo.id = res.id;
+          setTodos((prev) => [...prev, newTodo]);
+        }
+      })
+      .catch(() => setError('Unable to add a todo'))
+      .finally(() => setIsInputLocked(false)));
+  }, [tempTodos]);
+
   if (!USER_ID) {
     return <UserWarning />;
   }
 
+  const deleteTask = (id: number): void => {
+    setTodos((prev) => prev.filter((todo) => todo.id !== id));
+  };
+
+  const removeAllCompleted = () => {
+    todos
+      .filter((todo) => todo.completed)
+      .forEach((todo) => deleteTodo(todo.id)
+        .then(() => setTodos((prev) => prev
+          .filter((todoNew) => !todoNew.completed)))
+        .catch(() => setError('Unable to delete a todo')));
+  };
+
   return (
-    <section className="section container">
-      <p className="title is-4">
-        Copy all you need from the prev task:
-        <br />
-        <a href="https://github.com/mate-academy/react_todo-app-loading-todos#react-todo-app-load-todos">React Todo App - Load Todos</a>
-      </p>
-
-      <p className="subtitle">Styles are already copied</p>
-    </section>
+    <div className="todoapp">
+      <h1 className="todoapp__title">todos</h1>
+
+      <div className="todoapp__content">
+        {todos?.length && (
```

Comment:
This array will always exist
```suggestion
        {todos.length && (
```

Reply:
Try to refresh the page and see that there is 0 hanging as a footer. You have to either cast this todos.length to boolean or write `todos.length > 0`. This way you will have a boolean expression instead of 0 that won't get shown. I know we sometimes ask you to shorten the expressions, and it's better in general, but this is one case when you have to be more cautious with it
```suggestion
        {todos.length > 0 && (
```